modules/cpmso-yaml-failure-domain-vsphere.adoc

```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cpmso-configuration.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="cpmso-yaml-failure-domain-vsphere_{context}"]
+= Sample VMware vSphere failure domain configuration
+
+On VMware vSphere infrastructure, the infrastructure resource specification, `infrastructures.config.openshift.io`, defines failure domains for your cluster. The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains. A failure domain is an infrastructure resource that comprises a control plane machine set, a vCenter datacenter, vCenter datastore, and a network. 
+
+By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on hardware that is separate from the primary VMware vSphere infrastructure. A control plane machine set also balance control plane machines across defined failure domains to provide fault tolerance capabilities to your infrastructure.
+
+[NOTE]
+====
+If you modify the `ProviderSpec` configuration in the `ControlPlaneMachineSet` CR, the control plane machine set updates all control plane machines deployed on the primary infrastructure and each failure domain infrastructure.
+====
+
+:FeatureName: Defining a failure domain for a control plane machine set
+include::snippets/technology-preview.adoc[]
+
+.Example ProviderSpec configuration with specified failure domain names
+[source,yaml]
+----
+# ...
+template:
+  machineType: machines_v1beta1_machine_openshift_io
+  machines_v1beta1_machine_openshift_io:
+    failureDomains: <1>
+      platform: VSphere
+      vsphere: <2>
+      - name: <failure_domain_name1>
+      - name: <failure_domain_name2> 
+#...    
+----
+<1> A failure domain defines the vCenter location for {product-title} cluster nodes.
+<2> Defines failure domains by name for the control plane machine set.
+
+[IMPORTANT]
+====
+Each `failureDomains.platform.vsphere.name` field value in the `ControlPlaneMachineSet` CR must match the corresponding value defined in the `failureDomains.name` field of the infrastructure resource specification. Currently, the `vsphere.name` field is the only supported failure domain field that you can specify in the `ControlPlaneMachineSet` CR.
```

Comment:
I don't see infrastructure resource specification being used in the docs.

Reply:
:heavy_check_mark: